internal/keyrotation/server.go

```diff
@@ -19,14 +19,22 @@ import (
 	"context"
 	"fmt"
 	"net/http"
+	"sync"
 
-	"go.opencensus.io/trace"
-
-	"github.com/google/exposure-notifications-server/internal/logging"
 	revisiondb "github.com/google/exposure-notifications-server/internal/revision/database"
 	"github.com/google/exposure-notifications-server/internal/serverenv"
 )
 
+// Server hosts end points to manage key rotation
+type Server struct {
+	config     *Config
+	env        *serverenv.ServerEnv
+	revisionDB *revisiondb.RevisionDB
+
+	// mutex lock to be held over read/write operations when rotating keys.
+	mu sync.RWMutex
```

Comment:
I noted above, but I think we should acquire a table-level lock on revisionDB instead of relying on a mutex. A mutex isn't shared across instances.